File: content-solutions/stack/peek-record.java
import java.util.*;

class Solution {
    public List<Integer> solution(String[] commands) {
        Deque<Integer> stack = new ArrayDeque<>();
        List<Integer> answer = new ArrayList<>();
        for (String command : commands) {
            String[] parts = command.split(" ");
            if (parts[0].equals("push")) stack.push(Integer.parseInt(parts[1]));
            else if (parts[0].equals("pop")) {
                if (!stack.isEmpty()) stack.pop();
            } else answer.add(stack.isEmpty() ? -1 : stack.peek());
        }
        return answer;
    }
}

File: content-solutions/stack/peek-record.py
def solution(commands):
    stack = []
    answer = []
    for command in commands:
        parts = command.split()
        if parts[0] == "push":
            stack.append(int(parts[1]))
        elif parts[0] == "pop":
            if stack:
                stack.pop()
        else:
            answer.append(stack[-1] if stack else -1)
    return answer
